### PR TITLE
Added a new collection reverse-engineering-tools

### PR DESCRIPTION
**Pull request by KOLANICH**
- [x] I followed the contributing guidelines: https://github.com/github/explore/blob/master/CONTRIBUTING.md
- [x] I have no affiliation with the project I am suggesting (as a maintainer, creator, contractor, or employee).

### Which change are you proposing?

  - [ ] Suggesting edits to an existing topic or collection
  - [x] Curating a new topic or collection

---------------------------------------------------------------------
### Curating a new topic or collection

- [x] I've formatted my changes as a new folder directory, named for the topic or collection as it appears in the URL on GitHub (e.g. `https://github.com/topics/[NAME]` or `https://github.com/collections/[NAME]`)
- [ ] My folder contains a `*.png` image (if applicable) and `index.md`
- [x] All required fields in my `index.md` conform to the Style Guide and API docs: https://github.com/github/explore/tree/master/docs

---------------------------------------------------------------------

**Please note: we will close your PR without comment if you do not check the boxes above and provide ALL requested information.**

### CODE REVIEW

File: collections/reverse-engineering-tools/index.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  - java-decompiler/jd-gui
+  - 0xd4d/de4dot
+  - avast-tl/retdec
+  - yegord/snowman
+  - x64dbg/x64dbg
+  - mitmproxy/mitmproxy
+  - ohjeongwook/DarunGrim
+  - dbeaver/dbeaver
+  - lucasg/Dependencies
+display_name: Tools helping to understand applications and formats internals.
```
Comment (MikeMcQuaid) [inline collections/reverse-engineering-tools/index.md]: "Reverse Engineering Tools"?

File: collections/reverse-engineering-tools/index.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+---
+Get knowledge!
```
Comment (MikeMcQuaid) [inline collections/reverse-engineering-tools/index.md]: Could you expand these descriptions a bit? There's not much text here compared to other collections.

Comment (KOLANICH) [reply collections/reverse-engineering-tools/index.md]: this text was inspired by https://github.com/github/explore/blob/master/collections/software-development-tools/index.md#L17

File: collections/reverse-engineering-tools/index.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  - x64dbg/x64dbg
+  - mitmproxy/mitmproxy
+  - ohjeongwook/DarunGrim
+  - dbeaver/dbeaver
+  - lucasg/Dependencies
+  - aquynh/capstone
+  - beautify-web/js-beautify
+display_name: Reverse Engineering Tools
+---
+Get knowledge for fun and profit!
```
Comment (MikeMcQuaid) [inline collections/reverse-engineering-tools/index.md]: @KOLANICH Would still like to see it be a bit longer. https://en.wikipedia.org/wiki/Reverse_engineering#Reverse_engineering_of_software may provide some inspiration. Just a sentence or two would be good, thanks.

**Comment by rizalansari** [comment]
👍

File: collections/reverse-engineering-tools/index.md (hunk 3 above)
Comment (KOLANICH) [reply collections/reverse-engineering-tools/index.md]: I don't know what can be added without spoiling the laconicity.

Comment (MikeMcQuaid) [reply collections/reverse-engineering-tools/index.md]: The laconicity is a bug, not a feature 😉 

What about something like:

> Tools to aid with the reverse engineering of binary, minified or compiled computer programs or protocols.

Comment (KOLANICH) [reply collections/reverse-engineering-tools/index.md]: 1 tautology
2 not very correct. What is "minified or compiled  <...> protocols"?

Comment (MikeMcQuaid) [reply collections/reverse-engineering-tools/index.md]: @KOLANICH Please feel free to reword. I'm not really interested in arguing about this; let's either find something better or close this PR.

**Pull request closed without merging**